### PR TITLE
 Added correlation length feature.

### PR DESCRIPTION
**Pull request by AntonisCSt**
Changed engineering.py so it can include correlation length feature. I tested it out with Example notebook. It should work fine.

### CODE REVIEW

File: earthquakes/engineering.py [hunk 1]
```diff
@@ -8,7 +8,6 @@
 
 def sequence_generator(data, xcol="acoustic_data", ycol="time_to_failure", size=150000):
     """Generator that extracts segments of the signal from the data.
-
```
Comment (joepvdbogaert) [inline earthquakes/engineering.py]: You seem to have removed all the white space in the docstrings, which makes them a bit hard to read. Can you re-add the empty lines before `Parameters`, `Returns`, etc.?

File: earthquakes/engineering.py [hunk 2]
```diff
@@ -21,7 +20,6 @@ def sequence_generator(data, xcol="acoustic_data", ycol="time_to_failure", size=
     size: int, optional (default: 150,000),
         The number of observations to include in a single sequence. Should be left at
         its default to generate sequences similar to the test data.
-
```
Comment (joepvdbogaert) [inline earthquakes/engineering.py]: Here as well

File: earthquakes/engineering.py [hunk 3]
```diff
@@ -35,12 +33,32 @@ def sequence_generator(data, xcol="acoustic_data", ycol="time_to_failure", size=
...
+    z2 = np.array(acf) / (x_std**2)
+
+    def find_y_point(xa, xb, ya, yb, yc):
+        m = (ya - yb) / (xa - xb)
+        xc = yc / m - yb / m + xb
+        return xc
+
+    x1 = np.where(z2[0:50] == z2[0:50][z2[0:50] < 0.36][0])[0][0] - 1
+    x2 = np.where(z2[0:50] == z2[0:50][z2[0:50] < 0.36][0])[0][0]
+    return find_y_point(x1, x2, z2[0:50][x1], z2[0:50][x2], 0.368)
```
Comment (joepvdbogaert) [inline earthquakes/engineering.py]: It might be good to keep all our feature computations in the same `FeatureComputer` class. You can simply add this function as a method inside the class, e.g.:
```
class FeatureComputer():
    ...

    def _autocorr_length(x):
        # your method
```
And then call it in the `compute_features` method. Note that I added an underscore here to indicate that we won't be calling this function directly, but it is used internally by another method.

Next to that, I was wondering what the hardcoded values (e.g., 100, 50, 0.36, 0.368) mean in this function. Is it useful to vary these values, so should they be parameters of the function?

Comment (AntonisCSt) [reply earthquakes/engineering.py]: Thanks a lot, Joep!

I will change actually the:
 50->100
and the 0.36->0.368

It is not useful to vary the values as parameters of the function. Here is why:
The 100 value indicated that the function will calculate only the first 100 points of the Autocorrelated function-ACF.
This choice is for performance reasons. Obviously, if I calculate all the points it will take a lot of time. The length of the ACF is the same as the signals. For us the rest of the ACF because we want a value from the ACF function that is for sure in the first 100 points. That value is called the correlation length (ξ). 

and can be found (if ACF(x)=y) by ACF(ξ)=0.368.  The find_y_point function does this interpolation because the signal is discrete.

**To sum up**, I basically, I say that for sure there will be a ξ that satisfies the ACF(ξ)=0.368 in the first 100 points of the ACF that I calculate.

If you want more on why 0.368 I can provide you with some more mathematical blah blah links.

Comment (joepvdbogaert) [reply earthquakes/engineering.py]: Thanks for the explanation, I think I get the general idea. If you think it always holds that this value can be found in the first 100 points, then it's fine to do it this way. If there may be cases (perhaps in other competitions) in which this might not hold, I would still argue to make it an optional parameter with a default value of 100. This gives us at least the flexibility to change the value if necessary, without changing the code. I also say this because there are not really downsides in making it a parameter. But your call!

Comment (AntonisCSt) [reply earthquakes/engineering.py]: Thanks!

Yea that makes sense I 'll fix that also!
